airbyte-integrations/connectors/source-zuora/source_zuora/zuora_client.py

```diff
@@ -0,0 +1,270 @@
+        elif q_type == "show_tables":
+            base_query["query"] = "SHOW TABLES"
+        return base_query
+
+    # SUBMIT: data_query_job using data_query
+    def _submit_dq_job(self, dq_query: Dict) -> str:
+        submit_job = self._make_request(method="POST", url=f"{self.url_base}/query/jobs", payload=dq_query)
+        # self.logger.debug(submit_job.json()["data"]["id"])
+        return submit_job.json()["data"]["id"]
+
+    # CHECK: the submited data_query_job status
+    def _check_dq_job_status(self, dq_job_id: str) -> Dict:
+        # Define the job error statuses
+        errors = ["failed", "canceled", "aborted"]
+        # Error msg: the cursor_field cannot be resolved
+        cursor_error = "Column 'updateddate' cannot be resolved"
+        # Error msg: cannot process object
+        obj_read_error = "failed to process object"
+
+        status = None
+        success = "completed"
+        while status != success:
+            response = self._make_request(url=f"{self.url_base}/query/jobs/{dq_job_id}")
+            dq_job_check = response.json()
+            status = dq_job_check["data"]["queryStatus"]
+            if status in errors and cursor_error in dq_job_check["data"]["errorMessage"]:
+                raise ZOQLQueryFieldCannotResolve
+            elif status in errors and obj_read_error in dq_job_check["data"]["errorMessage"]:
+                raise ZOQLQueryCannotProcessObject
+            elif status in errors:
+                raise ZOQLQueryFailed(response)
+        return dq_job_check
+
+    # GET: data_query_job result
+    def _get_data_from_dq_job(self, response: requests.Response) -> List:
+        dq_job_data = requests.get(response["data"]["dataFile"])
+        dq_job_data.raise_for_status()
+        data = [j.loads(line) for line in dq_job_data.text.splitlines()]
+        return data or []
+
+    # Warpper function for `Query > Submit > Check > Get`
+    def _get_data(
+        self, q_type: str = "select", obj: str = None, date_field: str = None, start_date: str = None, end_date: str = None
+    ) -> Iterable:
+        query = self._make_dq_query(q_type, obj, date_field, start_date, end_date)
+        # self.logger.debug(f"{query}")
+        submit = self._submit_dq_job(query)
+        check = self._check_dq_job_status(submit)
+        get = self._get_data_from_dq_job(check)
+        yield from get
+
+    # Warper for _get_data using date-slices as pages
+    def _get_data_with_date_slice(
+        self, q_type: str, obj: str, date_field: str, start_date: str, window_days: int, end_date: str = None
+    ) -> Iterable:
+        # Parsing input dates
+        s = pendulum.parse(start_date)
+        tz = self._get_tz(s)
+        # If there is no `end_date` as input - use now()
+        e = pendulum.parse(pendulum.now().to_datetime_string()) if end_date is None else pendulum.parse(end_date)
+        # check end_date bigger than start_date
+        if s >= e:
```

Comment:
when could it happen? if it is data from config. then lets verify config before running all streams

Reply:
You're right, checked the code, this is the old test which is no more needed. Removed.